app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,60 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+
+  def perform(csv_export)
+    ActiveRecord::Base.connection_pool.with_connection do
+      create_export_folder(csv_export)
+      generate_csv(csv_export)
+      cleanup_downloaded
+    end
+  end
+
+  private
+
+  def cleanup_downloaded
+    CsvExport.old.each do |csv_export|
```

Comment:
Updated with an amended commit before I merged the master onto the branch, but undid the amend to make the fix clearer:)